python/tvm/topi/cuda/sparse.py

```diff
@@ -170,6 +170,16 @@ def gen_ir(data, w_data, w_indices, w_indptr, out):
         # TODO(tkonolige): seperate implementation for large block sizes
         ib = tvm.tir.ir_builder.create()
 
+        if tvm.target.Target.current(allow_none=False).kind.name == "rocm":
```

Comment:
I've never tested this kernel on vulkan, but since our vulkan and opencl target do not have a default `thread_warp_size` specified, I'm pretty sure trying to use warp instruction there wouldn't work. Moreover, even if we had a default `thread_warp_size` for vulkan for example, lowering warp instruction needs dedicated support from codegen that is not there for vulkan and opencl.

So for now, I think `use_warp_storage` should be True only for CUDA.

Reply:
I've defaulted to not using warp storage unless we are on cuda.